rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/util/ScanRecordParser.java

```diff
@@ -45,74 +34,37 @@
     private static final int DATA_TYPE_LOCAL_NAME_SHORT = 0x08;
     private static final int DATA_TYPE_LOCAL_NAME_COMPLETE = 0x09;
     private static final int DATA_TYPE_TX_POWER_LEVEL = 0x0A;
-    private static final int DATA_TYPE_SERVICE_DATA = 0x16;
+    private static final int DATA_TYPE_SERVICE_DATA_16_BIT = 0x16;
+    private static final int DATA_TYPE_SERVICE_DATA_32_BIT = 0x20;
+    private static final int DATA_TYPE_SERVICE_DATA_128_BIT = 0x21;
     private static final int DATA_TYPE_MANUFACTURER_SPECIFIC_DATA = 0xFF;
 
-    public static final ParcelUuid BASE_UUID =
-            ParcelUuid.fromString("00000000-0000-1000-8000-00805F9B34FB");
+    public static final UUID BASE_UUID =
+            UUID.fromString("00000000-0000-1000-8000-00805F9B34FB");
+
     /** Length of bytes for 16 bit UUID */
     public static final int UUID_BYTES_16_BIT = 2;
     /** Length of bytes for 32 bit UUID */
     public static final int UUID_BYTES_32_BIT = 4;
     /** Length of bytes for 128 bit UUID */
     public static final int UUID_BYTES_128_BIT = 16;
 
-    private static final String UUID_BASE_FORMAT = "%08x-0000-1000-8000-00805f9b34fb";
-
     @Inject
-    public UUIDUtil() {
+    public ScanRecordParser() {
```

Comment:
Unfortunately this could be an informal breaking change since I suggested people on StackOverflow to use this class as convenience :/ This is why I kept `UUIDUtil` so long.